src/lib/middleware.ts

```diff
@@ -1,89 +1,11 @@
 import { CookieSerializeOptions, serialize } from "cookie";
-import { IncomingHttpHeaders, IncomingMessage, ServerResponse } from "http";
-import {
-  GetServerSideProps,
-  GetServerSidePropsContext,
-  GetServerSidePropsResult,
-} from "next";
+import { IncomingMessage, ServerResponse } from "http";
+import { GetServerSidePropsContext } from "next";
 import { NextApiRequestCookies } from "next/dist/next-server/server/api-utils";
 import parse from "urlencoded-body-parser";
 import { v4 as uuidv4 } from "uuid";
-import { FormCacheFactory, FormSubmission, IFormCache } from "./formCache";
-import { Registration } from "./registration/registration";
-import { acceptRejectCookieId, formSubmissionCookieId } from "./types";
-import { toArray } from "./utils";
-
-export type BeaconsContext = GetServerSidePropsContext & {
-  showCookieBanner: boolean;
-  submissionId?: string;
-  formData: FormSubmission;
-  registration?: Registration;
-  useIndex: number;
-};
-
-export function withCookieRedirect<T>(callback: GetServerSideProps<T>) {
-  return async (
-    context: BeaconsContext
-  ): Promise<GetServerSidePropsResult<T>> => {
-    const cookies: NextApiRequestCookies = context.req.cookies;
-
-    if (!cookies || !cookies[formSubmissionCookieId]) {
-      return {
-        redirect: {
-          destination: "/",
-          permanent: false,
-        },
-      };
-    }
-
-    return callback(context);
-  };
-}
-
-/**
- * Decorator function to add beacons specific information to the `getServerSideProps` context.
- *
- * @param context {GetServerSidePropsContext}   The NextJS application context
- * @param addCacheFn {(context: BeaconsContext) => Promise<void>} (Optional) the function used to add the cache to the context
- * @returns       {Promise<BeaconsContext>}     A promise resolving to the decorated context containing application specific data
- */
-export async function decorateGetServerSidePropsContext(
-  context: GetServerSidePropsContext,
-  addCacheFn: (context: BeaconsContext) => Promise<void> = addCache
-): Promise<BeaconsContext> {
-  const decoratedContext: BeaconsContext = context as BeaconsContext;
-
-  addCookieBannerAcceptance(decoratedContext);
-  await addCacheFn(decoratedContext);
-  await addFormData(decoratedContext);
-  addRegistrationIndexes(decoratedContext);
-
-  return decoratedContext;
-}
```

Comment:
This and its composed functions are not required as the `Rule`s derive things like form data and the user's cookies from the `GetServerSidePropsContext` directly.